proj/SpeedLight/SpeedLight.ino

```diff
@@ -0,0 +1,62 @@
+/*
+ * Indicator Light for Motor Speed Setpoint:
+ * Measures angular velocity of encoder in degrees per second (moving average over every 250ms)
+ * and compares to arbitrary speed setpoint value. Lights up LED when difference is larger than arbitrary threshold.
+ */
+
+#define encoder0PinA  2
+#define encoder0PinB  3
+#define ledPin 13
+#define errorThreshold 20
+#define cpr 200 // Cycles per revolution of encoder disk
+
+int targetSpeed = 40; // Target angulary velocity / setpoint in degrees per second
```

Comment:
double?

Reply:
Didn't think decimal values were relevant for angular velocities, especially in this application because error threshold is likely to be significantly larger. Should I change it?

Reply:
I am not sure what values are you expecting?

Reply:
On that note, I might make a variable in the traditional m/s, then have another variable calculate it in angular velocity to make it easier for anyone coming later. Then make both variables doubles

Reply:
Considering that the encoder was rotating so fast that the arduino's interrupt frequency couldn't keep up during comp and a new board with an mbed is being designed to fix this, I expect that the speed values will be pretty large. And the error threshold that determines whether an LED warning light should go off will probably be a decently large value as well (more than several degrees per second), because if I'm not mistaken this system is meant to indicate when something is very wrong with the robot and human intervention is needed. Therefore I don't think storing the measured and target angular velocities as integers will be a problem.

Reply:
I see where you're coming from. Given the way the code is currently written, an variable of type int would do just fine. However, to have a variable in m/s would be more practical, and since a conversion from m/s to angular would certainly not be clean, both variables would have to be doubles.

Reply:
we also decreased the encoder to 48 ticks per revolution not 200. in terms of ticks per second you are looking at values of 50 - 114.

Reply:
Got it. For converting linear target speed to angular, should I define a wheel radius constant?

Reply:
Yes, you can find that constant in MotorShield.ino